lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -76,13 +105,13 @@ Function CheckModified-File($url, $dest, $headers, $credentials, $timeout, $use_
 
         $webLastMod = $webResponse.LastModified
     } Catch [System.Net.WebException] {
-        $result.status_code = $_.Exception.Response.StatusCode
-        Fail-Json -obj $result -message "Error requesting '$url'. $($_.Exception.Message)"
+        $module.Result.status_code = $_.Exception.Response.StatusCode
```

Comment:
I believe this is where the test issue is occurring, the move to C# means we are no longer using `ConvertTo-Json`. It seems like the PS cmdlet does a bit more to deal with things like PowerShell objects like a `PSParameterizedProperty` which `JavaScriptSerializer` falls flat on. To changing this to `$module.Result.status_code = $_.Exception.Response.StatusCode.ToString()` to ensure we are dealing with a native .NET type. I could be wrong as to whether it's `StatusCode` or maybe `TotalSeconds` used elsewhere so it's a matter of trying out what works. 

This is going to be a slight regression in features but we need `Ansible.Basic` to be compatibility with both native C# and .NET so we cannot use the PS library to do the work for us.

Reply:
I was suspecting that much, but I wasn't sure whether I would be doing to conversion, or this was something we wanted to fix in `Ansible.Basic`.

That said, we do cast to `int` a little bit below, so I think we can do this here as well. And if this fails, we should be looking into why that is. TBC